src/WebJobs.Script.WebHost/Controllers/InstanceController.cs

```diff
@@ -71,5 +71,23 @@ public IActionResult GetInstanceInfo()
         {
             return Ok(_instanceManager.GetInstanceInfo());
         }
+
+        [HttpPost]
+        [Route("admin/instance/disable")]
+        [Authorize(Policy = PolicyNames.AdminAuthLevel)]
+        public async Task<IActionResult> Disable([FromServices] IScriptHostManager hostManager)
+        {
+            if (_environment.IsLinuxContainerEnvironment())
+            {
+                _logger.LogInformation("Disabling container");
+                // Mark the container disabled. We check for this on host restart
+                await Utility.MarkContainerDisabled();
+                var unused = hostManager.RestartHostAsync();
```

Comment:
maybe do a Task.Run to run this in the background?

Reply:
following the same pattern in the restart api.

Reply:
is the concern with this taking too long to respond back?

Reply:
No, the concern is that this is somewhat of an anti-pattern, since if that Task throws, it will be unhandled and can bring down the process. https://markheath.net/post/async-antipatterns. See everywhere else in the code where we fire/forget async work. We use Task.Run, with exception handling

Reply:
Fixed.